.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,29 @@
+name: UnitTests
+
+on:
+  push:
+    branches: [ master, use_concurrent_ruby ]
+  pull_request:
+    branches: [ master, use_concurrent_ruby ]
```

Comment:
you don't need to mention here your branch name, master is sufficient. Otherwise, the workflow runs twice. You should mention here the branches you want run without a PR and the target branches of your PRs.